pkg/controller/admissionchecks/provisioning/provisioning.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	autoscaling "k8s.io/autoscaler/cluster-autoscaler/apis/provisioningrequest/autoscaling.x-k8s.io/v1beta1"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/util/admissioncheck"
+)
+
+func isProvisioned(pr *autoscaling.ProvisioningRequest) bool {
+	return apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Provisioned)
+}
+
+func isAccepted(pr *autoscaling.ProvisioningRequest) bool {
+	return apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Accepted)
+}
+
+func isFailed(pr *autoscaling.ProvisioningRequest) bool {
+	return apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Failed)
+}
+
+func isBookingExpired(pr *autoscaling.ProvisioningRequest) bool {
+	return apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.BookingExpired)
+}
+
+func isCapacityRevoked(pr *autoscaling.ProvisioningRequest) bool {
+	return apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.CapacityRevoked)
+}
+
+func ProvisioningRequestName(workloadName, checkName string, attempt int32) string {
+	fullName := fmt.Sprintf("%s-%s-%d", workloadName, checkName, int(attempt))
+	return limitObjectName(fullName)
+}
+
+func getProvisioningRequestNamePrefix(workloadName, checkName string) string {
+	fullName := fmt.Sprintf("%s-%s-", workloadName, checkName)
+	return limitObjectName(fullName)
+}
+
+func getProvisioningRequestPodTemplateName(prName, podsetName string) string {
+	fullName := fmt.Sprintf("%s-%s-%s", podTemplatesPrefix, prName, podsetName)
+	return limitObjectName(fullName)
+}
+
+func matchesWorkloadAndCheck(pr *autoscaling.ProvisioningRequest, workloadName, checkName string) bool {
+	attemptRegex := getAttemptRegex(workloadName, checkName)
+	matches := attemptRegex.FindStringSubmatch(pr.Name)
+	return len(matches) > 0
+}
+
+func getAttempt(log *logr.Logger, pr *autoscaling.ProvisioningRequest, workloadName, checkName string) int32 {
```

Comment:
```suggestion
func getAttempt(log logr.Logger, pr *autoscaling.ProvisioningRequest, workloadName, checkName string) int32 {
```

Logger is already an interface, so a pointer by itself.

Reply:
Fixed